tests/test_coreg/test_blockwise.py

```diff
@@ -147,6 +147,28 @@ def test_blockwise_coreg_pipeline(self, step, example_data, tmp_path, block_size
         valid = (expected.data.data != expected.nodata) & (aligned.data.data != aligned.nodata)
         assert np.allclose(expected.data.data[valid], aligned.data.data[valid], atol=20)
 
+    @pytest.mark.parametrize("block_size", [500, 985, 1332], ids=["2d_shifts", "1d_shifts_x", "monotile"])
+    def test_blockwise_coreg_pipeline_with_multiprocessing(self, step, example_data, tmp_path, block_size):
+        """Test end-to-end blockwise coregistration in multiprocessing and validate output."""
+        ref, tba, mask = example_data
+
+        config_mc = MultiprocConfig(
+            chunk_size=block_size, outfile=tmp_path / "test.tif", cluster=ClusterGenerator("multi", nb_workers=4)
+        )
+        blockwise_coreg = xdem.coreg.BlockwiseCoreg(step=step, mp_config=config_mc, block_size_fit=block_size)
+        blockwise_coreg.fit(ref, tba, mask)
+        blockwise_coreg.apply()
+
+        aligned = xdem.DEM(tmp_path / "aligned_dem.tif")
+
+        # Ground truth comparison with full image coregistration
+        nuth_kaab = xdem.coreg.NuthKaab()
+        expected = nuth_kaab.fit_and_apply(ref, tba, mask)
```

Comment:
This should use "step" instead, not necessarily NuthKaab?

Reply:
It will ensure full consistency. Actually, `step` currently uses `NuthKaab(vertical_shift=False)`, but not the NuthKaab redefined here.
This might explain why you have such drastic differences below for the `np.allclose()`